tests/system_tests/test_core/test_resume.py

```diff
@@ -99,3 +100,51 @@ def test_resume_output_log(wandb_backend_spy):
             if f.startswith("logs/output_") and f.endswith(".log")
         ]
         assert len(log_files) == 2
+
+
+def test_resume_config_preserves_image_mask(user, wandb_backend_spy):
+    img_array = np.zeros((2, 2, 3), dtype=np.uint8)
+    mask_array = np.zeros((1, 1), dtype=np.uint8)
+    mask_array[0, 0] = 1
+    class_labels = {1: "square"}
+
+    with wandb.init() as run:
+        run.log(
+            {
+                "test_image": wandb.Image(
+                    img_array,
+                    masks={
+                        "prediction": {
+                            "mask_data": mask_array,
+                            "class_labels": class_labels,
+                        }
+                    },
+                )
+            }
+        )
+
+    with wandb.init(id=run.id, resume="must") as run:
+        run.log(
+            {
+                "test_image_after_resume": wandb.Image(
+                    img_array,
+                    masks={
+                        "prediction": {
+                            "mask_data": mask_array,
+                            "class_labels": class_labels,
+                        }
+                    },
+                )
+            }
+        )
+
+    with wandb_backend_spy.freeze() as snapshot:
+        resumed_config = snapshot.config(run_id=run.id)
+        assert "_wandb" in resumed_config
+        resumed_wandb_config = resumed_config["_wandb"]["value"]
+        resumed_item_config = resumed_wandb_config.get("mask/class_labels", {})
```

Comment:
Can probably skip the assert as the `KeyError` would be self-explanatory, and no need to use `.get()` with a default for the `mask/class_labels` key since, if it's not there, the asserts later will fail anyway.

And for the name, maybe `class_labels` is better since it is shorter, and describes what it is expected to contain?

```suggestion
        class_labels = resumed_config["_wandb"]["value"]["mask/class_labels"]
```